### ISSUE TITLE
Random server disconnects for all players

### ISSUE THREAD

**Issue by RainOfPain125** [SmartlyDressedGames/Unturned-3.x-Community#380]
There seems to be no error or anything special in particular in log files or console. Multiple players will randomly get disconnected from the server in waves. My CPU and RAM is not being maxed out and have plenty of resources, my internet speed is consistent and can easily handle my servers. Why would this be happening? There is very little documentation online about AsyncTCPSocket but a lot of the answer say that it's usually just client-side people having bad internet. However, this is effecting multiple people and this happens for all of them at the same time.

Broadcast: [-]  [URSS-Sld-E1] Daniel
Disconnecting: PlayerID: 76561198192695189 Name: treep.garcia Character:  [URSS-Sld-E1] Daniel
Broadcast: [-] Master Maguku
Disconnecting: PlayerID: 76561198059553575 Name: te Character: Master Maguku
Broadcast: [-] Si se te cay el jabon llego yo:v
Disconnecting: PlayerID: 76561198251320814 Name: Olakace78 Character: Si se te cay el jabon llego yo:v
Broadcast: [-] kaboom
Disconnecting: PlayerID: 76561198114108266 Name: mrkremk Character: kaboom
Broadcast: [-] elZizar
Disconnecting: PlayerID: 76561198176529789 Name: elZizar Character: elZizar
Broadcast: [-] HydroGaming
Disconnecting: PlayerID: 76561198156501959 Name: ImNotHydro Character: HydroGaming
Broadcast: [-] Sergeant Key
Disconnecting: PlayerID: 76561198131422245 Name: Key Character: Sergeant Key
STEAMPS3 - AsyncTCPSocket created
STEAMPS3 - AsyncTCPSocket created
STEAMPS3 - AsncTCPSocket destroyed
STEAMPS3 - AsyncTCPSocket created
STEAMPS3 - AsyncTCPSocket created
STEAMPS3 - AsncTCPSocket destroyed
STEAMPS3 - AsyncTCPSocket created
STEAMPS3 - AsyncTCPSocket created
STEAMPS3 - AsyncTCPSocket created
STEAMPS3 - AsyncTCPSocket created
STEAMPS3 - AsyncTCPSocket created
STEAMPS3 - AsyncTCPSocket created

**Comment by N1GHTPATROL** [comment]
I believe this happens when you have too many servers running at the same time. Make sure your server is running on Performance Mode and try closing other servers if they are not necessary.

Even though your computer may be powerful, Unturned has its way of bottlenecking.

**Comment by RainOfPain125** [comment]
@N1GHTPATROL how would having multiple servers.. do such a thing? I really can't understand this one whatsoever.

And what do you mean by 'performance' mode?

**Comment by rube200** [comment]
I'm on windows if I select something and continue to select the server stops responding

**Comment by RainOfPain125** [comment]
@rube200 thats what happens when you click in console. It freezes the entire server. Just dont click in console, and if you do just right click on console to unfreeze or unselect.

**Comment by rube200** [comment]
from the moment that the white bar appears as if I had to select

**Comment by SDGNelson** [comment]
You're right that it always seem to happen at the same time as that STEAMPS3 - AsyncTCPSocket logging. When I've looked into it previously there were other devs who also use Steam's networking reporting the same issue, but no solution found. From some people I talked to it doesn't sound like there's anyone actively maintaining the older Steam features, so it's unlikely to be patched. Ideally Unturned would switch network APIs, but the game is too tightly built around SteamP2P. (at the moment 4 doesn't use Steam's networking)
